From 8f66862839387c909a25277fdc8eb1a8f79fbcac Mon Sep 17 00:00:00 2001
From: Michael Goffioul <michael.goffioul@lincor.com>
Date: Mon, 19 Apr 2021 13:34:09 -0400
Subject: [PATCH 1/3] Map flexible YUV format to NV12

The pixel format HAL_PIXEL_FORMAT_YCbCr_420_888 is used e.g. by WebView
when accessing a webcam (maybe from NdkImageReader?). Simply mapping the
flexible format to NV12 seems to do the trick on Baytrail platform.
---
 gralloc_gbm.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gralloc_gbm.cpp b/gralloc_gbm.cpp
index 5670213..d8ef7f3 100644
--- a/gralloc_gbm.cpp
+++ b/gralloc_gbm.cpp
@@ -98,6 +98,7 @@ static uint32_t get_gbm_format(int format)
 		break;
 	case HAL_PIXEL_FORMAT_YCbCr_422_SP:
 	case HAL_PIXEL_FORMAT_YCrCb_420_SP:
+	case HAL_PIXEL_FORMAT_YCbCr_420_888:
 	default:
 		fmt = 0;
 		break;
@@ -127,6 +128,7 @@ static int gralloc_gbm_get_bpp(int format)
 	case HAL_PIXEL_FORMAT_YV12:
 	case HAL_PIXEL_FORMAT_YCbCr_422_SP:
 	case HAL_PIXEL_FORMAT_YCrCb_420_SP:
+	case HAL_PIXEL_FORMAT_YCbCr_420_888:
 		bpp = 1;
 		break;
 	default:
@@ -507,6 +509,7 @@ int gralloc_gbm_bo_lock_ycbcr(buffer_handle_t handle,
 
 	switch (hnd->format) {
 	case HAL_PIXEL_FORMAT_YCrCb_420_SP:
+	case HAL_PIXEL_FORMAT_YCbCr_420_888:
 		ystride = cstride = GRALLOC_ALIGN(hnd->width, 16);
 		ycbcr->y = addr;
 		ycbcr->cr = (unsigned char *)addr + ystride * hnd->height;
-- 
2.25.1

